Improve navigation and clarify content.

brainwane opened the pull request:
- De-duplicate content that lives elsewhere and point to the current canonical source. (We're updating README.prod.md much more often than server.html.)
- Add link to zulip.com/api in top navbar.
- Add more guidance for new contributors.
- Fix several obsolete or unclear links.
- Fix formatting, whitespace, punctuation, and style.

Fixes: zulip/zulip#668.

brainwane committed:
Add more and better guidance for new contributors.

timabbott commented on server.html:
```diff
@@ -85,92 +92,10 @@
-              wget https://www.zulip.com/dist/releases/zulip-server-1.3.12.tar.gz<br />
-              echo "95f65a74f7447b5b4c46e7ed487fca50ff7a2c72 zulip-server-1.3.12.tar.gz" | sha1sum -c<br />
-              tar -xf zulip-server-1.3.12.tar.gz<br />
-              mv zulip-server-1.3.12 zulip<br />
```
Currently, this place is the only place where we document the sha1sums of the Zulip tarballs.  We need to put that somewhere before we can get rid of this...

brainwane replied:
Understood. I kept the sha1sums and got rid of the rest of the duplicated content.

timabbott commented on contribute.html:
```diff
@@ -47,40 +48,75 @@
+            an integration for your favorite service</a></li>
+            <li>Improve
+            our <a href="https://github.com/zulip/zulip/blob/master/Dockerfile">Dockerfile</a>
+            to make Zulip available to dockerized servers</li>
```
Would link to translations here, perhaps?

brainwane replied:
Addressed -- thanks.

timabbott commented on contribute.html:
```diff
@@ -47,40 +48,75 @@
+	<section>
+	  <h3>Security</h3>
+	  <p>Please report any security issues you discover to
+	  zulip-security@googlegroups.com.</p>
```
s/security issues/potential security issues/

timabbott commented:
I merged the 4 commits in this branch that I didn't have any comments on.  Let me know when you have the rest fixed up @brainwane !

timabbott committed:
Update Google analytics tracking code.

brainwane committed:
Suggest developers' instance for testing.

brainwane committed:
Deduplicate content from README.prod.md.

Right now, this is more nicely formatted than the README.prod.md,
but it's duplicative, and we're updating README.prod.md much
more frequently, which means server.html is out of date. Until
we've stabilized the division of Read the Docs/Zulip.org docs,
we'll only list requirements, the sha1sums of the Zulip tarballs,
and a few other minimal notes here, and point to README.prod.md
for installation instructions.

Fixes: zulip/zulip#668.

brainwane committed:
Fix word wrapping and header.

brainwane commented:
Thanks @timabbott -- the updated branch is better thanks to your comments, and ready to merge.

timabbott commented on contribute.html:
```diff
@@ -56,11 +56,21 @@
             <li>Add a service monitoring dashboard</li>
+	    <li><a href="https://zulip.readthedocs.io/en/latest/translating.html">Translate</a>
+	    user-visible words into another language</li>
             <li>Add your favorite emoji</li>
```
Let's get rid of emoji and dashboard and replace them with things like "contribute to the roadmap"

Or, thinking about this more, do you think we should just dump this page and make it a link to (potentially an improved version of) http://zulip.readthedocs.io/en/latest/readme-symlink.html#ways-to-contribute ?

brainwane replied:
I'll slim this page way down and link to http://zulip.readthedocs.io/en/latest/readme-symlink.html#ways-to-contribute -- good idea -- but I still think it's a good idea to have this as a standalone zulip.org page, one that's got our branding. And I like that we have an opportunity here to display the zulip-security@googlegroups.com address **on** zulip.org, where it's easy to find for someone who wants/needs it.

timabbott commented:
On server.html, what do you think about putting the "download" section above the "Installation instructions" section?  Seems like that might be cleaner.

brainwane committed:
Remove duplicate content from contributor guide.

brainwane commented:
@timabbott I tried it that way, and yup, you're right. Will do.

brainwane commented:
@timabbott Ready for you to review again.

timabbott commented:
Merged the server.html stuff!  Will read through the rest (and get Luke to deploy) later.  

One small note on language; I think throughout the project we've been using "development" rather than abbreviating to "dev", and I guess we should be consistent about that (though the place where you add use of "dev" is deleted in the next commit, so whatever).

brainwane commented:
@timabbott Aha, yes, I realize now that I didn't adhere to this convention; I'll be more careful in the future. Thanks.

timabbott commented:
Merged this.  I think we'll want to do more work on improving that contribute page; I don't think the list of project ideas is as good as it could be.  But it's definitely better than before.
